### ISSUE TITLE
Build fails on Fedora 37

### ISSUE THREAD

**Issue by mara004** [DaveMDS/python-efl#3]
```
python3 setup.py build
Python-EFL: 1.26.0
Checking for Python: OK, found 3.11.1
Checking for Cython: not needed, using pre-generated C files
Checking for Eina: OK, found 1.26.3
Checking for Eo: OK, found 1.26.3
Checking for Evas: OK, found 1.26.3
Checking for Ecore: OK, found 1.26.3
Checking for EcoreFile: OK, found 1.26.3
Checking for EcoreInput: OK, found 1.26.3
Checking for EcoreCon: OK, found 1.26.3
Checking for EcoreX: OK, found 1.26.3
Checking for Ethumb: OK, found 1.26.3
Checking for Ethumb_Client: OK, found 1.26.3
Checking for Edje: OK, found 1.26.3
Checking for Emotion: OK, found 1.26.3
Checking for DBus: OK, found 1.3.2
Checking for Elementary: OK, found 1.26.3
running build
running build_py
running build_ext
building 'efl.eo' extension
gcc -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -O2 -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -fPIC -I/usr/include/python3.11 -c efl/eo/efl.eo.c -o build/temp.linux-x86_64-cpython-311/efl/eo/efl.eo.o -I/usr/include/eo-1 -I/usr/include/eina-1 -I/usr/include/eina-1/eina -pthread -I/usr/include/valgrind -fno-var-tracking-assignments -Wno-misleading-indentation -Wno-deprecated-declarations -Wno-unused-variable -Wno-format-security
efl/eo/efl.eo.c: In function ‘__Pyx_PyErr_GetTopmostException’:
efl/eo/efl.eo.c:7417:21: error: ‘_PyErr_StackItem’ {aka ‘struct _err_stackitem’} has no member named ‘exc_type’
 7417 |     while ((exc_info->exc_type == NULL || exc_info->exc_type == Py_None) &&
      |                     ^~
efl/eo/efl.eo.c:7417:51: error: ‘_PyErr_StackItem’ {aka ‘struct _err_stackitem’} has no member named ‘exc_type’
 7417 |     while ((exc_info->exc_type == NULL || exc_info->exc_type == Py_None) &&
      |                                                   ^~
efl/eo/efl.eo.c: In function ‘__Pyx__ExceptionSave’:
efl/eo/efl.eo.c:7431:21: error: ‘_PyErr_StackItem’ {aka ‘struct _err_stackitem’} has no member named ‘exc_type’
 7431 |     *type = exc_info->exc_type;
      |                     ^~
efl/eo/efl.eo.c:7433:19: error: ‘_PyErr_StackItem’ {aka ‘struct _err_stackitem’} has no member named ‘exc_traceback’
 7433 |     *tb = exc_info->exc_traceback;
      |                   ^~
efl/eo/efl.eo.c: In function ‘__Pyx__ExceptionReset’:
efl/eo/efl.eo.c:7447:24: error: ‘_PyErr_StackItem’ {aka ‘struct _err_stackitem’} has no member named ‘exc_type’
 7447 |     tmp_type = exc_info->exc_type;
      |                        ^~
efl/eo/efl.eo.c:7449:22: error: ‘_PyErr_StackItem’ {aka ‘struct _err_stackitem’} has no member named ‘exc_traceback’
 7449 |     tmp_tb = exc_info->exc_traceback;
      |                      ^~
efl/eo/efl.eo.c:7450:13: error: ‘_PyErr_StackItem’ {aka ‘struct _err_stackitem’} has no member named ‘exc_type’
 7450 |     exc_info->exc_type = type;
      |             ^~
efl/eo/efl.eo.c:7452:13: error: ‘_PyErr_StackItem’ {aka ‘struct _err_stackitem’} has no member named ‘exc_traceback’
 7452 |     exc_info->exc_traceback = tb;
      |             ^~
efl/eo/efl.eo.c: In function ‘__Pyx__GetException’:
efl/eo/efl.eo.c:7534:28: error: ‘_PyErr_StackItem’ {aka ‘struct _err_stackitem’} has no member named ‘exc_type’
 7534 |         tmp_type = exc_info->exc_type;
      |                            ^~
efl/eo/efl.eo.c:7536:26: error: ‘_PyErr_StackItem’ {aka ‘struct _err_stackitem’} has no member named ‘exc_traceback’
 7536 |         tmp_tb = exc_info->exc_traceback;
      |                          ^~
efl/eo/efl.eo.c:7537:17: error: ‘_PyErr_StackItem’ {aka ‘struct _err_stackitem’} has no member named ‘exc_type’
 7537 |         exc_info->exc_type = local_type;
      |                 ^~
efl/eo/efl.eo.c:7539:17: error: ‘_PyErr_StackItem’ {aka ‘struct _err_stackitem’} has no member named ‘exc_traceback’
 7539 |         exc_info->exc_traceback = local_tb;
      |                 ^~
efl/eo/efl.eo.c: In function ‘__Pyx_AddTraceback’:
efl/eo/efl.eo.c:464:62: error: invalid use of incomplete typedef ‘PyFrameObject’ {aka ‘struct _frame’}
  464 |   #define __Pyx_PyFrame_SetLineNumber(frame, lineno)  (frame)->f_lineno = (lineno)
      |                                                              ^~
efl/eo/efl.eo.c:8885:5: note: in expansion of macro ‘__Pyx_PyFrame_SetLineNumber’
 8885 |     __Pyx_PyFrame_SetLineNumber(py_frame, py_line);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~
error: command '/usr/bin/gcc' failed with exit code 1

```

**Comment by mara004** [comment]
I also tried `ENABLE_CYTHON=1 python3 setup.py build` but it fails as well, though differently

```
Python-EFL: 1.26.0
Checking for Python: OK, found 3.11.1
Checking for Cython: OK, found 0.29.32
Checking for Eina: OK, found 1.26.3
Checking for Eo: OK, found 1.26.3
Checking for Evas: OK, found 1.26.3
Checking for Ecore: OK, found 1.26.3
Checking for EcoreFile: OK, found 1.26.3
Checking for EcoreInput: OK, found 1.26.3
Checking for EcoreCon: OK, found 1.26.3
Checking for EcoreX: OK, found 1.26.3
Checking for Ethumb: OK, found 1.26.3
Checking for Ethumb_Client: OK, found 1.26.3
Checking for Edje: OK, found 1.26.3
Checking for Emotion: OK, found 1.26.3
Checking for DBus: OK, found 1.3.2
Checking for Elementary: OK, found 1.26.3
Compiling efl/eo/efl.eo.pyx because it depends on /usr/lib64/python3.11/site-packages/Cython/Includes/libc/string.pxd.
Compiling efl/utils/conversions.pyx because it depends on /usr/lib64/python3.11/site-packages/Cython/Includes/libc/string.pxd.
Compiling efl/utils/logger.pyx because it depends on /usr/lib64/python3.11/site-packages/Cython/Includes/libc/string.pxd.
Compiling efl/evas/efl.evas.pyx because it depends on /usr/lib64/python3.11/site-packages/Cython/Includes/libc/string.pxd.
Compiling efl/ecore/efl.ecore.pyx because it depends on /usr/lib64/python3.11/site-packages/Cython/Includes/libc/string.pxd.
Compiling efl/ecore_input/efl.ecore_input.pyx because it depends on /usr/lib64/python3.11/site-packages/Cython/Includes/libc/string.pxd.
Compiling efl/ecore_con/efl.ecore_con.pyx because it depends on /usr/lib64/python3.11/site-packages/Cython/Includes/libc/string.pxd.
Compiling efl/ecore_x/efl.ecore_x.pyx because it depends on /usr/lib64/python3.11/site-packages/Cython/Includes/libc/string.pxd.
Compiling efl/ethumb/efl.ethumb.pyx because it depends on /usr/lib64/python3.11/site-packages/Cython/Includes/libc/string.pxd.
Compiling efl/ethumb/efl.ethumb_client.pyx because it depends on /usr/lib64/python3.11/site-packages/Cython/Includes/libc/string.pxd.
Compiling efl/edje/efl.edje.pyx because it depends on /usr/lib64/python3.11/site-packages/Cython/Includes/libc/string.pxd.
Compiling efl/edje_edit/efl.edje_edit.pyx because it depends on /usr/lib64/python3.11/site-packages/Cython/Includes/libc/string.pxd.
Compiling efl/emotion/efl.emotion.pyx because it depends on /usr/lib64/python3.11/site-packages/Cython/Includes/libc/string.pxd.
Compiling efl/dbus_mainloop/efl.dbus_mainloop.pyx because it depends on /usr/lib64/python3.11/site-packages/Cython/Includes/libc/string.pxd.
Compiling efl/elementary/__init__.pyx because it depends on /usr/lib64/python3.11/site-packages/Cython/Includes/libc/string.pxd.
[ 1/15] Cythonizing efl/elementary/__init__.pyx
[ 2/15] Cythonizing efl/evas/efl.evas.pyx
[ 3/15] Cythonizing efl/utils/conversions.pyx
[ 4/15] Cythonizing efl/dbus_mainloop/efl.dbus_mainloop.pyx
[ 5/15] Cythonizing efl/ecore/efl.ecore.pyx
[ 6/15] Cythonizing efl/ecore_con/efl.ecore_con.pyx
[ 7/15] Cythonizing efl/ecore_input/efl.ecore_input.pyx
[ 8/15] Cythonizing efl/ecore_x/efl.ecore_x.pyx
[ 9/15] Cythonizing efl/edje/efl.edje.pyx
[10/15] Cythonizing efl/edje_edit/efl.edje_edit.pyx
[11/15] Cythonizing efl/emotion/efl.emotion.pyx
[12/15] Cythonizing efl/eo/efl.eo.pyx
[13/15] Cythonizing efl/ethumb/efl.ethumb.pyx
[14/15] Cythonizing efl/ethumb/efl.ethumb_client.pyx
[15/15] Cythonizing efl/utils/logger.pyx
running build
running build_py
running build_ext
building 'efl.eo' extension
gcc -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -O2 -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -fPIC -I/usr/include/python3.11 -c efl/eo/efl.eo.c -o build/temp.linux-x86_64-cpython-311/efl/eo/efl.eo.o -I/usr/include/eo-1 -I/usr/include/eina-1 -I/usr/include/eina-1/eina -pthread -I/usr/include/valgrind -fno-var-tracking-assignments -Wno-misleading-indentation -Wno-deprecated-declarations -Wno-unused-variable -Wno-format-security
gcc -shared -Wl,-z,relro -Wl,--as-needed -Wl,-z,now -Wl,--build-id=sha1 -g -Wl,-z,relro -Wl,--as-needed -Wl,-z,now -Wl,--build-id=sha1 -g build/temp.linux-x86_64-cpython-311/efl/eo/efl.eo.o -L/usr/lib64 -o build/lib.linux-x86_64-cpython-311/efl/eo.cpython-311-x86_64-linux-gnu.so -leo -leina -pthread -lm -lrt -ldl
building 'efl.utils.deprecated' extension
creating build/temp.linux-x86_64-cpython-311/efl/utils
gcc -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -O2 -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -fPIC -I/usr/include/python3.11 -c efl/utils/deprecated.c -o build/temp.linux-x86_64-cpython-311/efl/utils/deprecated.o -I/usr/include/eina-1 -I/usr/include/eina-1/eina -pthread -I/usr/include/valgrind -fno-var-tracking-assignments -Wno-misleading-indentation -Wno-deprecated-declarations -Wno-unused-variable -Wno-format-security
efl/utils/deprecated.c: In function ‘__Pyx_AddTraceback’:
efl/utils/deprecated.c:461:62: error: invalid use of incomplete typedef ‘PyFrameObject’ {aka ‘struct _frame’}
  461 |   #define __Pyx_PyFrame_SetLineNumber(frame, lineno)  (frame)->f_lineno = (lineno)
      |                                                              ^~
efl/utils/deprecated.c:4806:5: note: in expansion of macro ‘__Pyx_PyFrame_SetLineNumber’
 4806 |     __Pyx_PyFrame_SetLineNumber(py_frame, py_line);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~
error: command '/usr/bin/gcc' failed with exit code 1
```

**Comment by mara004** [comment]
Similar error with clang

```
efl/utils/deprecated.c:4806:5: error: incomplete definition of type 'struct _frame'
    __Pyx_PyFrame_SetLineNumber(py_frame, py_line);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
efl/utils/deprecated.c:461:62: note: expanded from macro '__Pyx_PyFrame_SetLineNumber'                                                                                                        
  #define __Pyx_PyFrame_SetLineNumber(frame, lineno)  (frame)->f_lineno = (lineno)
                                                      ~~~~~~~^
/usr/include/python3.11/pytypedefs.h:22:16: note: forward declaration of 'struct _frame'                                                                                                      
typedef struct _frame PyFrameObject;
               ^
1 error generated.                                                                                                                                                                            
error: command '/usr/bin/clang' failed with exit code 1
```

**Comment by mara004** [comment]
https://bugzilla.redhat.com/show_bug.cgi?id=2042360 sounds related
Apparently this is a Cython <-> Python3.11 interaction issue...

**Comment by mara004** [comment]
With the Cython 3 alpha release, this happens
```
[me@fedora python-efl-1.26.0]$ CC=clang ENABLE_CYTHON=1 python3 setup.py build
Python-EFL: 1.26.0
Checking for Python: OK, found 3.11.1
Checking for Cython: OK, found 3.0.0a11
Checking for Eina: OK, found 1.26.3
Checking for Eo: OK, found 1.26.3
Checking for Evas: OK, found 1.26.3
Checking for Ecore: OK, found 1.26.3
Checking for EcoreFile: OK, found 1.26.3
Checking for EcoreInput: OK, found 1.26.3
Checking for EcoreCon: OK, found 1.26.3
Checking for EcoreX: OK, found 1.26.3
Checking for Ethumb: OK, found 1.26.3
Checking for Ethumb_Client: OK, found 1.26.3
Checking for Edje: OK, found 1.26.3
Checking for Emotion: OK, found 1.26.3
Checking for DBus: OK, found 1.3.2
Checking for Elementary: OK, found 1.26.3
Traceback (most recent call last):
  File "/home/me/Downloads/builds/python-efl-1.26.0/setup.py", line 303, in <module>
    ext_modules = cythonize(
                  ^^^^^^^^^^
  File "/home/me/.local/lib/python3.11/site-packages/Cython/Build/Dependencies.py", line 985, in cythonize
    module_list, module_metadata = create_extension_list(
                                   ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/me/.local/lib/python3.11/site-packages/Cython/Build/Dependencies.py", line 834, in create_extension_list
    kwds = deps.distutils_info(file, aliases, base).values
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/me/.local/lib/python3.11/site-packages/Cython/Build/Dependencies.py", line 687, in distutils_info
    return (self.transitive_merge(filename, self.distutils_info0, DistutilsInfo.merge)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/me/.local/lib/python3.11/site-packages/Cython/Build/Dependencies.py", line 696, in transitive_merge
    return self.transitive_merge_helper(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/me/.local/lib/python3.11/site-packages/Cython/Build/Dependencies.py", line 708, in transitive_merge_helper
    for next in outgoing(node):
                ^^^^^^^^^^^^^^
  File "Cython/Utils.py", line 129, in Cython.Utils.cached_method.wrapper
  File "/home/me/.local/lib/python3.11/site-packages/Cython/Build/Dependencies.py", line 609, in cimported_files
    pxd_file = self.find_pxd(module, filename)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "Cython/Utils.py", line 129, in Cython.Utils.cached_method.wrapper
  File "/home/me/.local/lib/python3.11/site-packages/Cython/Build/Dependencies.py", line 590, in find_pxd
    pxd = self.context.find_pxd_file(relative, source_file_path=filename)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/me/.local/lib/python3.11/site-packages/Cython/Compiler/Main.py", line 245, in find_pxd_file
    pxd = self.search_include_directories(
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/me/.local/lib/python3.11/site-packages/Cython/Compiler/Main.py", line 273, in search_include_directories
    return search_include_directories(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "Cython/Utils.py", line 90, in Cython.Utils.cached_function.wrapper
  File "/home/me/.local/lib/python3.11/site-packages/Cython/Compiler/Main.py", line 669, in search_include_directories
    warning(pos, "Dotted filenames ('%s') are deprecated."
  File "/home/me/.local/lib/python3.11/site-packages/Cython/Compiler/Errors.py", line 215, in warning
    listing_file = threadlocal.cython_errors_listing_file
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: '_thread._local' object has no attribute 'cython_errors_listing_file'
```

**Comment by mara004** [comment]
Don't know why/how, but building worked after replacing the `DEPRECATED` class with this stub:
```python
class DEPRECATED:

    def __init__(self, version, message):
        self.version = version
        self.message = message

    def __call__(self, func_or_meth):
        return func_or_meth
```

Of course, this disables deprecation warnings as a drawback...

**Comment by DaveMDS** [comment]
This should be fixed now in 1.26.1 release
